### PR TITLE
EM-820: Implement GitHub Actions CI/CD pipeline for vanilla-components to verify build

### PR DESCRIPTION
**Pull request by yashrajk17**
# Description

This commit sets up a CI/CD pipeline for vanilla-components. Upon each push to the main branch, it installs dependencies, builds embeddable components, and alerts of any errors. It runs on Ubuntu, macOS, and Windows machines for broad compatibility.

# Main Changes

- Implemented GitHub Actions workflow YAML file to establish CI/CD pipeline.
- Defined triggers to activate the pipeline upon pushing changes to the main branch.
- Incorporated steps to checkout the main branch, install dependencies using npm, and build embeddable components.
- Added error-checking mechanism to notify users of any encountered issues during the execution.
- Configured the pipeline to run on Ubuntu, macOS, and Windows machines, ensuring compatibility across various environments.

### CODE REVIEW

File: .github/workflows/verify-build.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Test on ${{ matrix.os }}
+        run: echo "Testing on ${{ matrix.os }}"
+
+      - name: Checkout main branch
+        uses: actions/checkout@v2
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 20.x
```
Comment (baseilos) [inline .github/workflows/verify-build.yml:26]: Can we make it into a parameter?

File: .github/workflows/verify-build.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Install dependencies
+        run: npm i
+
+      - name: Build embeddable
+        run: npm run embeddable:build
+
+      - name: Check for errors
```
Comment (baseilos) [inline .github/workflows/verify-build.yml:34]: Is this step needed? Won't it automatically fail when some of previous command fails?

**Review by baseilos — CHANGES_REQUESTED** [review]

File: .github/workflows/verify-build.yml (hunk 2 above)
Comment (yashrajk17) [reply .github/workflows/verify-build.yml:34]: Yes we can remove it

File: .github/workflows/verify-build.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      - main
+
+jobs:
+  verify-build:
+    name: Verify Build for Vanilla Components
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        node_version: [20.x]
```
Comment (baseilos) [inline .github/workflows/verify-build.yml:15]: Let's run it for node 18, 19, 20.

**Review by baseilos — CHANGES_REQUESTED** [review]

**Review by baseilos — APPROVED** [review]

**Pull request merged**